src/backends/arch.rs

```diff
@@ -90,12 +99,27 @@ impl Arch {
         no_confirm: bool,
         config: &Config,
     ) -> Result<()> {
-        run_args(
+        run_command(
             [self.command, "--remove", "--recursive"]
                 .into_iter()
                 .chain(config.aur_rm_args.iter().map(String::as_str))
                 .chain(Some("--no_confirm").filter(|_| no_confirm))
                 .chain(packages.keys().map(String::as_str)),
+            Perms::AsRoot,
         )
     }
+
+    pub fn try_parse_toml_package(
+        &self,
+        toml: &toml::Value,
+    ) -> Result<(ArchPackageId, ArchInstallOptions)> {
+        match toml {
+            toml::Value::String(x) => Ok((x.to_string(), Default::default())),
+            toml::Value::Table(x) => Ok((
+                x.clone().try_into::<StringPackageStruct>()?.package,
```

Comment:
Slightly confused here. Where are the two traits implemented?

Reply:
The `try_into()` is an inherent trait on the `toml::Table` struct to skip going back to`String` before calling `toml::from_str()`. The `StringPackageStruct` is a bodge so that we can extract just the `.package` field from the toml table if it is a `String`, which is the case for all the backends at the moment.